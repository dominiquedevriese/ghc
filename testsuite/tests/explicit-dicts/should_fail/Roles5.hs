{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE RoleAnnotations #-}
{-# LANGUAGE TypeApplications #-}
module Roles5 where

import GHC.Base (Ord(..))

type role Set nominal
data Set a

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

f5 = (undefined :: Ord Int => Int -> Set Int -> Set Int)
     ((reverseOrd @Int :: Ord Int))
-- Also flagged by coherence check
