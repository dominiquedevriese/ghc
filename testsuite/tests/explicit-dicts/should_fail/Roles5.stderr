
Roles5.hs:14:6: error:
    • Explicit dictionary application not allowed
    • In the expression:
        (undefined :: Ord Int => Int -> Set Int -> Set Int)
          ((reverseOrd @Int :: Ord Int))
      In an equation for ‘f5’:
          f5
            = (undefined :: Ord Int => Int -> Set Int -> Set Int)
                ((reverseOrd @Int :: Ord Int))
